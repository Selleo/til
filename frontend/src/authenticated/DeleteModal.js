import React, { useState, useEffect } from 'react'
import Modal from 'react-modal'

const DeleteModal = props => {
  const [isModalOpen, setIsModalOpen] = useState(true)
  const { deletePost, toggleModal } = props

  /* backdrop-filter: blur(2px); */
  const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-90%',
      transform: 'translate(-50%, -50%)',
      backgroundColor: '#272B2F',
      border: 'none',
      width: '576px',
    },
  }
  useEffect(() => {
    Modal.setAppElement('#root')
  }, [])

  const handleDelete = () => {
    deletePost()
    setIsModalOpen(false)
    toggleModal()
  }

  const handleCancel = () => {
    setIsModalOpen(false)
    toggleModal()
  }

  return (
    <Modal
      isOpen={isModalOpen}
      style={customStyles}
      shouldCloseOnOverlayClick={true}
      overlayClassName="Overlay"
    >
      <h3 className="delete-modal__header">
        Are you sure you want to delete this post?
      </h3>
      <div className="delete-modal">
        <button className="cancel-button" onClick={handleCancel}>
          Cancel
        </button>
        <button onClick={handleDelete} className="delete-post-btn">
          Delete
        </button>
      </div>
    </Modal>
  )
}

export default DeleteModal
