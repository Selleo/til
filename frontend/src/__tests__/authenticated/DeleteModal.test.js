import React from "react";
import { render, cleanup } from "@testing-library/react";
import Modal from "react-modal";
import DeleteModal from "../../authenticated/DeleteModal";

describe("<DeleteModal/>", () => {
  beforeEach(() => {
    const modalContainer = document.createElement("div");
    modalContainer.id = "root";
    document.body.appendChild(modalContainer);

    Modal.setAppElement(modalContainer);
  });

  afterEach(cleanup);

  it("renders correctly", () => {
    const { container } = render(<DeleteModal />);

    expect(container).toMatchSnapshot();
  });
});
