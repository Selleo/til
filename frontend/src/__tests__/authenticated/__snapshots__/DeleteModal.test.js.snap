// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteModal/> renders correctly 1`] = `<div />`;
